packages/contracts-bedrock/scripts/upgrades/holocene/.env.example

```diff
@@ -0,0 +1,74 @@
+##############################################
+#               ↓  Required  ↓               #
+##############################################
+
+# The network to deploy the contracts to.
+# Must be one of 'mainnet', 'sepolia'
+NETWORK=
+
+# Etherscan API key used to verify contract bytecode
+ETHERSCAN_API_KEY=
+
+# RPC URL for the L1 network that matches $NETWORK
+ETH_RPC_URL=
+
+# Private key used to deploy the new contracts for this upgrade
+PRIVATE_KEY=
+
+# Path to the deploy config JSON file
+DEPLOY_CONFIG_PATH=
+
+# Path to the folder where output artifacts will be stored
+OUTPUT_FOLDER_PATH=
+
+# Address of deployed `PreimageOracle` contract.
+PREIMAGE_ORACLE_ADDR=
+
+# Address of deployed `AnchorStateRegistry` proxy contract.
+ANCHOR_STATE_REGISTRY_PROXY_ADDR=
+
+# Address of the `SuperchainConfig` proxy contract.
+SUPERCHAIN_CONFIG_PROXY_ADDR=
+
+# Address of deployed `ProxyAdmin` contract.
+PROXY_ADMIN_ADDR=
+
+# Address of deployed `SystemConfig` proxy contract.
+SYSTEM_CONFIG_PROXY_ADDR=
+
+# Address of deployed `DisputeGameFactory` proxy contract.
+DISPUTE_GAME_FACTORY_PROXY_ADDR=
+
+###################################################
+#                 ↓  Optional  ↓                  #
+# Do not set if you don't know what you're doing. #
+###################################################
+
+# Address of the deployed `SystemConfig` implementation for Holocene.
+#
+# This implementation is reused across L2 deployments based on the L1 @ `ETH_RPC_URL`.
+# If you are not the first to deploy Holocene on this L1, this field should be set to
+# the existing deployment address.
+#
+# If this field is not set, the `superchain-registry` will be consulted for the implementation address.
+# If this field is set to the zero address, a new `SystemConfig` implementation will be deployed.
+SYSTEM_CONFIG_IMPL_ADDR=
+
+# Address of the deployed `MIPS` implementation for Holocene.
+#
+# This implementation is reused across L2 deployments based on the L1 @ `ETH_RPC_URL`.
+# If you are not the first to deploy Holocene on this L1, this field should be set to
+# the existing deployment address.
+#
+# If this field is not set, the `superchain-registry` will be consulted for the implementation address.
+# If this field is set to the zero address, a new `MIPS` implementation will be deployed.
+MIPS_IMPL_ADDR=
+
+# Address of deployed `DelayedWETH` proxy contract.
+#
+# If this field is not set, a new `DelayedWETH` proxy will be deployed, and used for the new
+# `FaultDisputeGame` and `PermissionedDisputeGame` deployments.
```

Comment:
Note: We should have separate DelayedWETH proxies for `FaultDisputeGame` and `PermissionedDisputeGame`. They can share the same implementation but should be different proxies so the bonds for each game type is held in separate contracts (we switched to this as part of Granite).

Reply:
Yeah the deployment script is setting both game types to the same DelayedWETH which is incorrect.  I don't think the superchain registry has ever caught up to the fact that there's two separate DelayedWETH now. The suggestion was that it should only track the game implementation contracts and everything downstream of that should just be loaded by querying those implementations but not sure where that work is up to - it still seems to track one of the two DelayedWETH proxies I think.

Reply:
For now, what I did:
1. Optional `DelayedWETH` implementation address in `.env`
    - If set to `address(0)`, a new implementation will be deployed.
    - If set to any other address, that address will be used as the proxy implementations.
    - If unset, the `delayed_weth` implementation address will be pulled from the `superchain-registry` for the network.
2. Unconditionally, new `DelayedWETH` proxies will be deployed for the new dispute game implementations.

This has a bit of a downside of using new proxies, but omits extra configuration where chain operators would need to supply their existing proxies or deploy new ones. If we're going this route of not re-using the same `DelayedWETH` proxy across games, we might not want to proxy it anymore to reduce a bit of operational overhead.

LMK if this fits your ask.

Reply:
> it still seems to track one of the two DelayedWETH proxies I think.

Yeah. Though tracking the implementation the two (or more soon) proxies use is beneficial, ideally they can all re-use the same implementation contract.

Reply:
Yeah I'm ok with that. `op-dispute-mon` will automatically pick up and monitor the new DelayedWETH proxy instances because it always loads them from the dispute games and all the funds will wind up being claimed back so we can then forget about the old ones.  I think we would want to keep them proxied though in case there's a bug in DelayedWETH - we can then upgrade the implementation and not lose any bonds still in the contract.